api/clients/v2/disperser_client.go

```diff
@@ -390,7 +393,126 @@ func (c *disperserClient) GetPaymentState(ctx context.Context) (*disperser_rpc.G
 		Signature: signature,
 		Timestamp: timestamp,
 	}
-	return c.client.GetPaymentStateForAllQuorums(ctx, request)
+	allQuorumsReply, err := c.client.GetPaymentStateForAllQuorums(ctx, request)
+	if err != nil {
+		// Check if error is "method not found" or "unimplemented"
+		if isMethodNotFoundError(err) {
+			// Fall back to old method
+			return c.getPaymentStateFromLegacyAPI(ctx, accountID, signature, timestamp)
+		}
+		return nil, err
+	}
+
+	return allQuorumsReply, nil
+}
+
+// this is true if we are targetting a disperser that hasn't upgraded to the new API yet.
+func isMethodNotFoundError(err error) bool {
+	if st, ok := status.FromError(err); ok {
+		return st.Code() == codes.Unimplemented
+	}
+	return false
+}
+
+func (c *disperserClient) getPaymentStateFromLegacyAPI(
+	ctx context.Context, accountID gethcommon.Address, signature []byte, timestamp uint64,
+) (*disperser_rpc.GetPaymentStateForAllQuorumsReply, error) {
+	// Convert old request/response format to new format
+	oldRequest := &disperser_rpc.GetPaymentStateRequest{
```

Comment:
Is the idea that `GetPaymentStateRequest` is entirely deprecated now? If that's the case, it would be really good to have a document there

Actually, now that I look at it more, why is there a new struct at all? `GetPaymentStateRequest` and `GetPaymentStateForAllQuorums` seem to have all the same fields. cc @hopeyen

Reply:
It does have the same fields. I wanted to keep the names consistent and later remove `GetPaymentStateRequest`

Reply:
Makes sense. This would be really good to document on the deprecated request.